src/lib/components/SelectField/SelectField.jsx

```diff
@@ -177,15 +191,18 @@ SelectField.propTypes = {
   layout: PropTypes.oneOf(['horizontal', 'vertical']),
   /**
    * Set of options to be chosen from.
+   *
+   * Both option or optgroup are accepted.
    */
-  options: PropTypes.arrayOf(PropTypes.shape({
-    disabled: PropTypes.bool,
-    label: PropTypes.string.isRequired,
-    value: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.number,
-    ]),
-  })).isRequired,
+  options: PropTypes.oneOfType([
+    PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.string,
```

Comment:
Label [mandatory according to W3C](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup#attr-label).